### PR TITLE
Issue 655 - ability to configure cluster SecurityGroup

### PR DESCRIPTION
**Pull request by martin-ducar-gd**
### What does this PR do?

This to me is first improvement in resolving https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/655

### Motivation

Bug in security group creation when no node groups security group is created

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### Additional Notes
Before:
Log described in issue https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/655
After:
```sh
Initializing modules...
Initializing the backend...
Initializing provider plugins...
- Reusing previous version of terraform-aws-modules/http from the dependency lock file
- Reusing previous version of hashicorp/kubernetes from the dependency lock file
- Reusing previous version of hashicorp/tls from the dependency lock file
- Reusing previous version of hashicorp/cloudinit from the dependency lock file
- Reusing previous version of hashicorp/aws from the dependency lock file
- Reusing previous version of hashicorp/null from the dependency lock file
- Reusing previous version of hashicorp/random from the dependency lock file
- Reusing previous version of hashicorp/helm from the dependency lock file
- Reusing previous version of kbst/kustomization from the dependency lock file
- Reusing previous version of gavinbunney/kubectl from the dependency lock file
- Reusing previous version of hashicorp/local from the dependency lock file
- Using previously-installed kbst/kustomization v0.5.0
- Using previously-installed gavinbunney/kubectl v1.14.0
- Using previously-installed hashicorp/local v2.2.3
- Using previously-installed terraform-aws-modules/http v2.4.1
- Using previously-installed hashicorp/tls v3.4.0
- Using previously-installed hashicorp/cloudinit v2.2.0
- Using previously-installed hashicorp/null v3.1.1
- Using previously-installed hashicorp/kubernetes v2.11.0
- Using previously-installed hashicorp/aws v4.2.0
- Using previously-installed hashicorp/random v3.3.1
- Using previously-installed hashicorp/helm v2.6.0
╷
│Warning: Experimental feature "module_variable_optional_attrs" is active
│
│  on .terraform/modules/eks/modules/launch-templates/locals.tf line 4, in terraform:
│   4:   experiments = [module_variable_optional_attrs]
│
│Experimental features are subject to breaking changes in future minor or
│patch releases, based on feedback.
│
│If you have feedback on the design of this feature, please open a GitHub
│issue to discuss it.
╵
Terraform has been successfully initialized!
You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.
If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
╷
│Warning: Experimental feature "module_variable_optional_attrs" is active
│
│  on .terraform/modules/eks/modules/launch-templates/locals.tf line 4, in terraform:
│   4:   experiments = [module_variable_optional_attrs]
│
│Experimental features are subject to breaking changes in future minor or
│patch releases, based on feedback.
│
│If you have feedback on the design of this feature, please open a GitHub
│issue to discuss it.
╵
Outputs:
region = "us-east-1"
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
+ create
<= read (data resources)
Terraform will perform the following actions:
  # module.eks.data.aws_eks_cluster.cluster[0] will be read during apply
  # (config refers to values not yet known)
<= data "aws_eks_cluster" "cluster"  {
+arn                       = (known after apply)
+certificate_authority     = (known after apply)
+created_at                = (known after apply)
+enabled_cluster_log_types = (known after apply)
+endpoint                  = (known after apply)
+id                        = (known after apply)
+identity                  = (known after apply)
+kubernetes_network_config = (known after apply)
+name                      = (known after apply)
+platform_version          = (known after apply)
+role_arn                  = (known after apply)
+status                    = (known after apply)
+tags                      = (known after apply)
+version                   = (known after apply)
+vpc_config                = (known after apply)
    }
  # module.eks.data.http.eks_cluster_readiness[0] will be read during apply
  # (config refers to values not yet known)
<= data "http" "eks_cluster_readiness"  {
+body             = (known after apply)
+ca_certificate   = (known after apply)
+id               = (known after apply)
+response_headers = (known after apply)
+timeout          = 600
+url              = (known after apply)
    }
  # module.eks.kubernetes_config_map.aws_auth[0] will be created
+ resource "kubernetes_config_map" "aws_auth" {
+data = {
+"mapAccounts" = jsonencode([])
+"mapRoles"    = jsonencode([])
+"mapUsers"    = jsonencode([])
        }
+id   = (known after apply)
+metadata {
+generation       = (known after apply)
+labels           = {
+"app.kubernetes.io/created-by" = "terraform-aws-eks-blueprints"
+"app.kubernetes.io/managed-by" = "terraform-aws-eks-blueprints"
            }
+name             = "aws-auth"
+namespace        = "kube-system"
+resource_version = (known after apply)
+uid              = (known after apply)
        }
    }
  # module.eks.module.aws_eks.data.tls_certificate.this[0] will be read during apply
  # (config refers to values not yet known)
<= data "tls_certificate" "this"  {
+certificates = (known after apply)
+id           = (known after apply)
+url          = (known after apply)
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["charge-code"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "charge-code"
+resource_id = (known after apply)
+value       = "<redacted>"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "<redacted>"
+resource_id = (known after apply)
+value       = "eks-node"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["data-classification"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "data-classification"
+resource_id = (known after apply)
+value       = "green"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["entity"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "entity"
+resource_id = (known after apply)
+value       = "<redacted>"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["environment"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "environment"
+resource_id = (known after apply)
+value       = "dev"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "<redacted>"
+resource_id = (known after apply)
+value       = "<redacted>"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["product-name"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "product-name"
+resource_id = (known after apply)
+value       = "<redacted>"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "<redacted>"
+resource_id = (known after apply)
+value       = "<redacted>"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["tenancy"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "tenancy"
+resource_id = (known after apply)
+value       = "multi"
    }
  # module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["used_for"] will be created
+ resource "aws_ec2_tag" "cluster_primary_security_group" {
+id          = (known after apply)
+key         = "used_for"
+resource_id = (known after apply)
+value       = "non_prod"
    }
  # module.eks.module.aws_eks.aws_eks_cluster.this[0] will be created
+ resource "aws_eks_cluster" "this" {
+arn                       = (known after apply)
+certificate_authority     = (known after apply)
+created_at                = (known after apply)
+enabled_cluster_log_types = [
+"api",
+"audit",
+"controllerManager",
+"scheduler",
        ]
+endpoint                  = (known after apply)
+id                        = (known after apply)
+identity                  = (known after apply)
+name                      = "demo23"
+platform_version          = (known after apply)
+role_arn                  = "<redacted>"
+status                    = (known after apply)
+tags                      = {
+"charge-code"            = "<redacted>"
+"<redacted>" = "eks-node"
+"data-classification"    = "green"
+"entity"                 = "<redacted>"
+"environment"            = "dev"
+"<redacted>"             = "<redacted>"
+"product-name"           = "<redacted>"
+"<redacted>"             = "<redacted>"
+"tenancy"                = "multi"
+"used_for"               = "non_prod"
        }
+tags_all                  = {
+"charge-code"            = "<redacted>"
+"<redacted>" = "eks-node"
+"data-classification"    = "green"
+"entity"                 = "<redacted>"
+"environment"            = "dev"
+"<redacted>"             = "<redacted>"
+"product-name"           = "<redacted>"
+"<redacted>"             = "<redacted>"
+"tenancy"                = "multi"
+"used_for"               = "non_prod"
        }
+version                   = "1.21"
+encryption_config {
+resources = [
+"secrets",
            ]
+provider {
+key_arn = "<redacted>"
            }
        }
+kubernetes_network_config {
+ip_family         = "ipv4"
+service_ipv4_cidr = (known after apply)
        }
+timeouts {
+create = "60m"
+delete = "30m"
+update = "60m"
        }
+vpc_config {
+cluster_security_group_id = (known after apply)
+endpoint_private_access   = true
+endpoint_public_access    = false
+public_access_cidrs       = [
+"0.0.0.0/0",
            ]
+security_group_ids        = (sensitive)
+subnet_ids                = (sensitive)
+vpc_id                    = (known after apply)
        }
    }
  # module.eks.module.aws_eks.aws_iam_openid_connect_provider.oidc_provider[0] will be created
+ resource "aws_iam_openid_connect_provider" "oidc_provider" {
+arn             = (known after apply)
+client_id_list  = [
+"sts.amazonaws.com",
        ]
+id              = (known after apply)
+tags            = {
+"Name"                   = "demo23-eks-irsa"
+"charge-code"            = "<redacted>"
+"<redacted>" = "eks-node"
+"data-classification"    = "green"
+"entity"                 = "<redacted>"
+"environment"            = "dev"
+"<redacted>"             = "<redacted>"
+"product-name"           = "<redacted>"
+"<redacted>"             = "<redacted>"
+"tenancy"                = "multi"
+"used_for"               = "non_prod"
        }
+tags_all        = {
+"Name"                   = "demo23-eks-irsa"
+"charge-code"            = "<redacted>"
+"<redacted>" = "eks-node"
+"data-classification"    = "green"
+"entity"                 = "<redacted>"
+"environment"            = "dev"
+"<redacted>"             = "<redacted>"
+"product-name"           = "<redacted>"
+"<redacted>"             = "<redacted>"
+"tenancy"                = "multi"
+"used_for"               = "non_prod"
        }
+thumbprint_list = (known after apply)
+url             = (known after apply)
    }
Plan: 13 to add, 0 to change, 0 to destroy.
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Creating...
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [1m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [1m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [1m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [1m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [1m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [1m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [2m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [2m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [2m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [2m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [2m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [2m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [3m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [3m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [3m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [3m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [3m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [3m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [4m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [4m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [4m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [4m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [4m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [4m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [5m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [5m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [5m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [5m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [5m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [5m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [6m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [6m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [6m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [6m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [6m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [6m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [7m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [7m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [7m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [7m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [7m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [7m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [8m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [8m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [8m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [8m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [8m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [8m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [9m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [9m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [9m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [9m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [9m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [9m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [10m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [11m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [11m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [11m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [11m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [11m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [11m50s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [12m0s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [12m10s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [12m20s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [12m30s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Still creating... [12m40s elapsed]
module.eks.module.aws_eks.aws_eks_cluster.this[0]: Creation complete after 12m44s [id=demo23]
module.eks.data.aws_eks_cluster.cluster[0]: Reading...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["entity"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["used_for"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["tenancy"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["charge-code"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["environment"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["data-classification"]: Creating...
module.eks.data.aws_eks_cluster.cluster[0]: Read complete after 0s [id=demo23]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["product-name"]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"]: Creation complete after 0s [id=sg-09d00b438fd171d87,product_id]
module.eks.module.aws_eks.data.tls_certificate.this[0]: Reading...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["used_for"]: Creation complete after 0s [id=sg-09d00b438fd171d87,used_for]
module.eks.data.http.eks_cluster_readiness[0]: Reading...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["data-classification"]: Creation complete after 0s [id=sg-09d00b438fd171d87,data-classification]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["charge-code"]: Creation complete after 0s [id=sg-09d00b438fd171d87,charge-code]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"]: Creation complete after 0s [id=sg-09d00b438fd171d87,compliance:na:defender]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["environment"]: Creation complete after 0s [id=sg-09d00b438fd171d87,environment]
module.eks.module.aws_eks.data.tls_certificate.this[0]: Read complete after 0s [id=<redacted>]
module.eks.module.aws_eks.aws_iam_openid_connect_provider.oidc_provider[0]: Creating...
module.eks.data.http.eks_cluster_readiness[0]: Read complete after 0s [id=<redacted>]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["tenancy"]: Creation complete after 0s [id=<redacted>]
module.eks.kubernetes_config_map.aws_auth[0]: Creating...
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["product-name"]: Creation complete after 0s [id=<redacted>]
module.eks.module.aws_eks.aws_iam_openid_connect_provider.oidc_provider[0]: Creation complete after 0s [id=<redacted>]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["<redacted>"]: Creation complete after 0s [id=<redacted>]
module.eks.module.aws_eks.aws_ec2_tag.cluster_primary_security_group["entity"]: Creation complete after 1s [id=sg-09d00b438fd171d87,entity]
╷
│Warning: Experimental feature "module_variable_optional_attrs" is active
│
│  on .terraform/modules/eks/modules/launch-templates/locals.tf line 4, in terraform:
│   4:   experiments = [module_variable_optional_attrs]
│
│Experimental features are subject to breaking changes in future minor or
│patch releases, based on feedback.
│
│If you have feedback on the design of this feature, please open a GitHub
│issue to discuss it.
╵
╷
│Error: Post "http://localhost/api/v1/namespaces/kube-system/configmaps": dial tcp 127.0.0.1:[80](https://github.com/McK-Internal/CNG_CCP-terraform_bootstrapper/runs/6996917909?check_suite_focus=true#step:3:81): connect: connection refused
│
│  with module.eks.kubernetes_config_map.aws_auth[0],
│  on .terraform/modules/eks/aws-auth-configmap.tf line 1, in resource "kubernetes_config_map" "aws_auth":
│   1: resource "kubernetes_config_map" "aws_auth" {
│
╵
Error: Process completed with exit code 1.
```

Error in auth is not related to topic and is caused by missing provider kubernetes config

### CODE REVIEW

**Comment by bryantbiggs** [comment]
@martin-ducar-gd  what are you solving for - what is the outcome you are trying to achieve?

**Comment by martin-ducar-gd** [comment]
To be able to deploy cluster with security groups ideally for both nodes and controll plane (optional but simple to achive) to be managed outside of module.

**Comment by martin-ducar-gd** [comment]
Currently controll plane SGs are not configurable with the nodes security groups on our side to be managed outside as must have requirement and with the bug mentioned it makes it unusable. Simply adding the controll plane SGs customization solves the whole issue for us.

**Comment by bryantbiggs** [comment]
Being able to disable the security group that the EKS module creates for the control plane (changing the hardcoded `true` to a variable) is perfectly acceptable. However, I think theres a few things to clarify here:
1. When you create an EKS cluster, the EKS service is going to automatically create a security group. This will now show up in IaC, its created by the service. This security group allows all inter-security group traffic for resources that have that security group attached and it allows all outbound access. When you using Fargate profiles, this is the security group that is used and that is not configurable. When you use the default launch template created by EKS managed node groups, that is the security group used
2. The security group that the EKS module creates is the bare minimum access required to deploy a cluster and reach a healthy state. You can read more about it [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/network_connectivity.md#security-groups)
3. There is the variable `cluster_additional_security_group_ids` to attach externally created security groups - which I believe is what you are after (partially)
4. We provide the node-group security group and cluster security group with the ability to add additional rules so that users can reference either security group in the additional rules they might add 

That said - there are no paths currently where the cluster primary security group can be provided by users. The EKS service will always create this and set it as the primary security group. The only options users have is to provide additional security groups as well as control over when the EKS service created security group is attached (except for Fargate profiles, that is again controlled by the service)

So we can accept a PR that will allow users to enable/disable the module created cluster security group, but this will not control the primary security group

**Review by vara-bonthu — APPROVED** [review]
LGTM 👍🏼

**Comment by martin-ducar-gd** [comment]
@bryantbiggs I understand but we use custom launch template due to modifications first second we don't use Fargate profiles now. So I get your concern however this way we have ability to don't produce one arbitrary security group one in this managed nodegroup scenario with custom launch template. Is it ok to merge?

**Review by bryantbiggs — APPROVED** [review]
yes of course, I just wanted to be clear that based on your information that this might not reach the resolution you are after. but the change itself is ok, we can merge

File: variables.tf [hunk 1]
```diff
@@ -54,6 +54,18 @@ variable "cluster_version" {
...
+variable "create_cluster_security_group" {
+  description = "Toggle to create or assign cluster security group"
+  type        = bool
+  default     = true
+}
+
+variable "cluster_security_group_id" {
+  description = "Security group to be used if creation of cluster security group is turned off"
+  type        = string
+  default     = ""
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply variables.tf:66]: it doesn't have to be - either an empty string or null will work here

**Pull request merged**